### PR TITLE
enhancement(logging): clean logs, don't throw on URL

### PR DESCRIPTION
**Pull request by sidvishnoi**

### CODE REVIEW

File: utils/logging.js [hunk 1]
```diff
@@ -75,11 +85,12 @@ const skipCommon = (req, res) => {
   const { method, path, query } = req;
   const { statusCode } = res;
   const ref = req.get("referer") || req.get("referrer");
-  const referrer = ref ? new URL(ref) : null;
```
Comment (sidvishnoi) [inline utils/logging.js]: saw someone send a request with `referfer: 'example.com'` which crashed the server

File: utils/logging.js [hunk 2]
```diff
@@ -33,14 +33,24 @@ const prettyJSON = (() => {
...
 /** @type {import('morgan').FormatFn} */
 const formatter = (tokens, req, res) => {
   const date = tokens.date(req, res, "iso");
   const remoteAddr = tokens["remote-addr"](req, res);
   const method = tokens.method(req, res);
   const status = parseInt(tokens.status(req, res), 10);
-  const url = new URL(tokens.url(req, res), "https://respec.org/");
-  const referrer = tokens.referrer(req, res);
+  /** @type {URL|null} */
+  const url = $try(() => new URL(tokens.url(req, res), "https://respec.org/"));
```
Comment (marcoscaceres) [inline utils/logging.js]: I.... ok... I wonder if this should just be URL specific tho... a generic $try seems like trouble when we should just try/catch things.

Comment (sidvishnoi) [reply utils/logging.js]: `tryURL(string)`?

File: utils/logging.js (hunk 1 above)
Comment (marcoscaceres) [reply utils/logging.js]: I want a referfer tattoo.

**Pull request merged**